ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/layered/LayeredRemovalReason.java

```diff
@@ -50,37 +45,53 @@ enum RemovedFrom {
   }
 
   /** The reason why the tx has been removed from the pool */
-  enum PoolRemovalReason implements RemovalReason {
-    /** Tx removed since it is confirmed on chain, as part of an imported block. */
-    CONFIRMED,
-    /** Tx removed since it has been replaced by another one added in the same layer. */
-    REPLACED,
-    /** Tx removed since it has been replaced by another one added in another layer. */
-    CROSS_LAYER_REPLACED,
-    /** Tx removed when the pool is full, to make space for new incoming txs. */
-    DROPPED,
+  enum PoolRemovalReason implements LayeredRemovalReason {
+    /**
+     * Tx removed since it is confirmed on chain, as part of an imported block. Keep tracking since
+     * makes no sense to reprocess a confirmed.
+     */
+    CONFIRMED(false),
+    /**
+     * Tx removed since it has been replaced by another one added in the same layer. Keep tracking
+     * since makes no sense to reprocess a replaced tx.
+     */
+    REPLACED(false),
+    /**
+     * Tx removed since it has been replaced by another one added in another layer. Keep tracking
+     * since makes no sense to reprocess a replaced tx.
+     */
+    CROSS_LAYER_REPLACED(false),
+    /**
+     * Tx removed when the pool is full, to make space for new incoming txs. Stop tracking it so we
+     * could re-accept it in the future.
+     */
+    DROPPED(true),
     /**
      * Tx removed since found invalid after it was added to the pool, for example during txs
-     * selection for a new block proposal.
+     * selection for a new block proposal. Keep tracking since we do not want to reprocess an
+     * invalid tx.
      */
-    INVALIDATED,
+    INVALIDATED(false),
     /**
      * Special case, when for a sender, discrepancies are found between the world state view and the
      * pool view, then all the txs for this sender are removed and added again. Discrepancies, are
      * rare, and can happen during a short windows when a new block is being imported and the world
-     * state being updated.
+     * state being updated. Keep tracking since it is removed and re-added.
      */
-    RECONCILED,
+    RECONCILED(false),
     /**
      * When a pending tx is penalized its score is decreased, if at some point its score is lower
-     * than the configured minimum then the pending tx is removed from the pool.
+     * than the configured minimum then the pending tx is removed from the pool. Stop tracking it so
+     * we could re-accept it in the future.
      */
-    BELOW_MIN_SCORE;
+    BELOW_MIN_SCORE(true);
```

Comment:
there is no added risk here, because a malicious actor can already send a bunch of those txs and the layered txpool already has protection for this, moreover in this PR the tx is removed from the pool and the possibly re-added, so not increasing the count, so this path is definitely more complex, that just sending new tx.

Reply:
@garyschulte do you need any other clarification/detail?